### DISCUSSION TITLE
Is there a way to serialize selection along with editorState?

### DISCUSSION THREAD

**Discussion by ChunxuYang** [facebook/lexical#5357] [Q&A] [answered]
Selection is part of editor state, I want to restore the selection from database when reloading.

However, when serializing editor states using `toJSON()`, lexical omits the selection. So when using `parseEditorState()`, the restored state's `_selection` falls to null.

Is there a way to save and restore the selection in database?

**Comment by yf-yang** [comment, marked as answer]
Lexical doesn't support it (Seems you've read the source code, so no need to explain that).
To ser/deser selection, the idea is pretty similar to what @lexical-yjs does (yjs is a state synchronization package, if you've used collaborative editor, you know that you are able to see other's cursor when editing your parts, that's what it do), so maybe you can refer to that package.
I'd recommend you to consider manage that state yourself. For example, register an update listener, whenever the selection changes, sync the state to your own state management module. When you serialize the state, just serialize your own states as well.
